app/services/ems_storage_dashboard_service.rb

```diff
@@ -90,4 +95,36 @@ def heatmaps
       :title         => 'Used capacity [%]'
     }
   end
+
+  def agg_events
+    event_hash = {}
+    event_list = []
+    events = EmsEvent.where("ems_id=#{@ems_id}")
+
+    events.each do |event|
+      unless event_hash[event.physical_storage_name]
+        event_hash[event.physical_storage_name] = {:fixed => 0, :not_fixed => 0}
+      end
+      if event.event_type == "autosde_critical_alert_fixed"
+        event_hash[event.physical_storage_name][:fixed] = event_hash[event.physical_storage_name][:fixed] + 1
```

Comment:
Ruby has `+=` so you could do `event_hash[event.physical_storage_name][:fixed] += 1`